filesystem/src/main/java/com/nytimes/android/external/fs/BufferedSourceAdapter.java

```diff
@@ -0,0 +1,16 @@
+package com.nytimes.android.external.fs;
+
+import javax.annotation.Nonnull;
+
+import okio.BufferedSource;
+
+/**
+ * Converts Java values to JSON, and JSON values to Java.
+ */
+public interface BufferedSourceAdapter<Parsed> {
+    @Nonnull
+    BufferedSource toJson(@Nonnull Parsed value);
+
+    @Nonnull
+    Parsed fromJson(@Nonnull BufferedSource source);
```

Comment:
I was going off the last comment on the issue when I added `fromJson` here, but I actually don't think it's needed since the Store's `Parser` implementation will handle this. I'll pull it out 👍